### PR TITLE
Refactoring rcmdcheck constructor

### PR DESCRIPTION
**Pull request by hadley**
This series of commits substantially simplifies `new_rcmdcheck()` so that object creation is more consistent through rcmdcheck.

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -5,7 +5,6 @@ S3method(print,rcmdcheck)
 S3method(print,rcmdcheck_comparison)
 S3method(summary,rcmdcheck)
 S3method(summary,rcmdcheck_comparison)
-export(check_details)
```
Comment (gaborcsardi) [inline NAMESPACE]: I think we need this. AFAIR it is used in `revdepcheck` as well.

**Comment by gaborcsardi** [comment]
I don't understand what's going on on Travis. Does it pick up the "package" that is included for testing? 

https://travis-ci.org/r-lib/rcmdcheck/jobs/265877071#L1116

File: NAMESPACE (hunk 1 above)
Comment (hadley) [inline NAMESPACE]: I can un-remove it so we don't break the revdeps, but I think it's better to have a stricter definition of the object itself and then just use that directly.

**Comment by hadley** [comment]
Assuming wifi is good on the plane, I'll fix, rebase and test in a couple of hours.

**Comment by hadley** [comment]
I'm going to merge this because I've heard people are having problems with revdepcheck so I want to get all my API fixes in in one pass.

**Comment by gaborcsardi** [comment]
The revdepcheck problems were fixed already, @krlmlr fixed them.

**Comment by hadley** [comment]
Strangely enough I didn't see the errors, even before pulling in his fixes

**Pull request merged**